File: backend/fertilar-service/src/main/java/com/fertilar/service/entity/Sensor.java
package com.fertilar.service.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Table(name = "sensores")
public class Sensor {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pila_id", nullable = false)
    private Pila pila;

    @Column(nullable = false, unique = true, length = 100)
    private String codigo;

    @Column(nullable = false, length = 100)
    private String tipo;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @Column(nullable = false)
    private Boolean activo = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: frontend/fertilar-ui/src/types/sensor.ts
export type SensorResumen = {
  id: string
  pilaId: string
  codigo: string
  tipo: string
  activo: boolean
}

export type Sensor = SensorResumen & {
  descripcion: string | null
  createdAt: string
  updatedAt: string
}

export type SensorRequest = {
  pilaId: string
  codigo: string
  tipo: string
  descripcion?: string
  activo: boolean
}
